content/blog/accessing-dtap-in-pods.md

```diff
@@ -0,0 +1,258 @@
+
+PySpark is an interface for Apache Spark in Python. Apache Spark is a unified analytics engine for big data processing, with built-in modules for streaming, SQL, machine learning and graph processing. Apache Spark can access data from HDFS and, with extension, file systems managed by DataTap. 
+
+## Install ```pyspark```
+
+There are lots of way to install Spark. The simplest way is to install pyspark package directly using ```pip install pyspark```. Run the following to install the prerequisite packages and pyspark.
+
+```bash
+# install pyspark & Java
+apt-get install python3 -y
+apt-get install python3-pip -y
+DEBIAN_FRONTEND=noninteractive apt-get install openjdk-11-jdk-headless -y
+pip install pyspark
+```
+
+There are two ways to interact with pyspark. The first one is to execute the ```pyspark``` command in bash to initiate the pyspark session. The second way is that to treat pyspark as a module which ```python``` kernel can import to. (```import pyspark```)
+
+### Method one: initiate ```pyspark``` session with jars
+In order to use datatap with pyspark, you have to add an external jar file as argument to pyspark. Initiate Spark's interactive shell in python using the following command. 
+
+```bash
+# bash
+
+# Specify the path of the jars files
+pyspark --jars /opt/bdfs/bluedata-dtap.jar
+```
+
+After starting the interactive shell, ```Spark Context``` and ```Spark Session``` are automatically initiate for you.
+
+![image](https://user-images.githubusercontent.com/72959956/120170783-e8d00680-c233-11eb-9fe8-136da9996fdc.png)
+
+We have to specify the Hadoop configurations. After that, you can read files from DataTap just like normally what you did with HDFS.
+
+```py
+# pyspark
+
+# Specify the Hadoop configurations.
+sc._jsc.hadoopConfiguration().set('fs.dtap.impl', 'com.bluedata.hadoop.bdfs.Bdfs')
+sc._jsc.hadoopConfiguration().set('fs.AbstractFileSystem.dtap.impl', 'com.bluedata.hadoop.bdfs.BdAbstractFS')
+
+# Commands for reading DataTap file.
+text = sc.textFile("dtap://TenantStorage/HPE.txt")
+text.take(5)
+```
+
+![image](https://user-images.githubusercontent.com/72959956/120171213-61cf5e00-c234-11eb-8928-2514e8b867a8.png)
+
+### Method two: initiate ```python``` and initiate ```pyspark``` with jars at runtime
+
+Run the Python Shell first:
+
+```bash
+# bash
+python3
+```
+
+At the Python runtime, add the path of the jar file using Spark configuration command:
+
+```py
+# python
+from pyspark import SparkConf, SparkContext
+
+# Specify the path of the jars files
+conf = SparkConf().set("spark.jars", "/opt/bdfs/bluedata-dtap.jar")
+sc = SparkContext( conf=conf)
+
+# Specify the Hadoop configurations.
+sc._jsc.hadoopConfiguration().set('fs.dtap.impl', 'com.bluedata.hadoop.bdfs.Bdfs')
+sc._jsc.hadoopConfiguration().set('fs.AbstractFileSystem.dtap.impl', 'com.bluedata.hadoop.bdfs.BdAbstractFS')
+
+# Commands for reading DataTap file.
+text = sc.textFile("dtap://TenantStorage/HPE.txt")
+text.take(5)
+```
+
+> References: 
```

Comment:
Because there are 2 items in the references, I suggest you use a "dash" or "asterix" to make the references a list of items. (make sure to insert a line break before and after each list item). :-)